src/main/java/edu/kis/powp/jobs2d/drivers/monitoring/DriverMonitorDecorator.java

```diff
@@ -2,42 +2,90 @@
 
 import edu.kis.powp.jobs2d.drivers.AbstractDecorator;
 import edu.kis.powp.jobs2d.drivers.VisitableJob2dDriver;
+import edu.kis.powp.jobs2d.drivers.monitoring.DriverEventManager.DriverEventType;
 import edu.kis.powp.observer.Publisher;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
 
 public class DriverMonitorDecorator extends AbstractDecorator {
     private final DriverUsageMonitor monitor;
     private final Publisher movePublisher = new Publisher();
-    private static boolean monitorEnabled = true;
-    public DriverMonitorDecorator(VisitableJob2dDriver driver, DriverUsageMonitor monitor) {
+    private final DriverMonitoringConfig config;
+    private final DriverLimitValidator validator;
+
+    private static boolean globalMonitorEnabled = true;
+
+    public DriverMonitorDecorator(VisitableJob2dDriver driver, DriverUsageMonitor monitor, DriverMonitoringConfig config) {
         super(driver);
-        this.monitor = monitor;
+        this.monitor = Objects.requireNonNull(monitor, "Monitor cannot be null");
+        this.config = config != null ? config : createDefaultConfig();
+        this.validator = new DriverLimitValidator(
+                this.config.getDriverParameters(),
+                this.config.getEventManager()
+        );
```

Comment:
DriverLimitValidator should not be in DriverMonitorDecorator (which should remain simple).
> This class should be only to update monitoring status. 

I think that DriverLimitValidator should be triggered by DriverUsageMonitor (e.g., best by using observer).